woocommerce/payment-gateway/payment-tokens/class-sv-wc-payment-gateway-payment-tokens-handler.php

```diff
@@ -584,6 +584,35 @@ public function update_tokens( $user_id, $tokens, $environment_id = null ) {
 	}
 
 
+	/**
+	 * Updates a single legacy token in user meta.
+	 *
+	 * @internal
+	 *
+	 * @since 5.6.0-dev.1
+	 *
+	 * @param int $user_id WP user ID
+	 * @param SV_WC_Payment_Gateway_Payment_Token $token token to update
+	 * @param string|null $environment_id optional environment ID, defaults to plugin current environment
+	 * @return string|int updated user meta ID
+	 */
+	public function update_legacy_token( $user_id, $token, $environment_id = null ) {
```

Comment:
Howe about keeping this helper as private inside the `SV_WC_Payment_Gateway_Payment_Token` object? I see you are fetching the payment tokens handler from the gateway object. Perhaps you could delegate [this](https://github.com/skyverge/wc-plugin-framework/blob/50f57745f9d3cfee1570ed3c1bdb27f29a42d3db/woocommerce/payment-gateway/payment-tokens/class-sv-wc-payment-gateway-payment-token.php#L584-L587) to the helper method itself?

Reply:
This helper method uses `get_environment_id` and `get_user_meta_name` from the handler and is very similar to the existing `update_tokens` (that will be changed soon). I do believe it makes sense for it to live on the handler... Do you believe it would be better for it to live in SV_WC_Payment_Gateway_Payment_Token object and retrieve the handler to call these methods?

Reply:
fair enough, let's keep it as you propose - we can always review this later if we feel to  -- thanks